internal/controller/siteconfig_controller.go

```diff
@@ -135,10 +139,17 @@ func (r *SiteConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return res, err
 	}
 
+	if !r.isSiteConfigValidated(siteConfig) {
```

Comment:
What if the siteconfig is updated? Wouldn't we want to validate every time?

Reply:
I will update the `isSiteConfigValidated` in a subsequent PR to keep validating until the cluster is provisioned. For now, this PR aims to introduce the siteconfig validation.